compatibility.h

```diff
@@ -0,0 +1,134 @@
+/* -------------------------------------------------------------------------
+ *
+ * compatibility.h
+ *
+ * Definitions for maintaining compatibility across Postgres versions.
+ *
+ * Copyright (c) 2010-2020, PostgreSQL Global Development Group
+ *
+ * -------------------------------------------------------------------------
+ */
+
+#ifndef SET_USER_COMPAT_H
+#define SET_USER_COMPAT_H
+
+/*
+ * PostgreSQL version 13+
+ *
+ * Introduces QueryCompletion struct
+ */
+#if PG_VERSION_NUM >= 130000
+static void PU_hook(PlannedStmt *pstmt, const char *queryString,
+					ProcessUtilityContext context, ParamListInfo params,
+					QueryEnvironment *queryEnv,
+					DestReceiver *dest, QueryCompletion *qc);
+#define _PU_HOOK \
+static void PU_hook(PlannedStmt *pstmt, const char *queryString, \
+					ProcessUtilityContext context, ParamListInfo params, \
+					QueryEnvironment *queryEnv, \
+					DestReceiver *dest, QueryCompletion *qc)
+
+#define _prev_hook \
+	prev_hook(pstmt, queryString, context, params, queryEnv, dest, qc)
+
+#define _standard_ProcessUtility \
+	standard_ProcessUtility(pstmt, queryString,	context, params, queryEnv, dest, qc)
+
+#endif /* 13+ */
+
+/*
+ * PostgreSQL version 12+
+ *
+ * - Removes OID column
+ */
+#if PG_VERSION_NUM >= 120000
+#define HEAP_TUPLE_GET_OID
+static inline Oid
+_heap_tuple_get_oid(HeapTuple roleTup)
+{
+	return ((Form_pg_authid) GETSTRUCT(roleTup))->oid;
+}
+
+#endif /* 12+ */
+
+/*
+ * PostgreSQL version 10+
+ *
+ * - Introduces PlannedStmt struct
+ * - Introduces varlena.h
+ */
+#if PG_VERSION_NUM >= 100000
+# ifndef _PU_HOOK
+static void PU_hook(PlannedStmt *pstmt, const char *queryString,
+					ProcessUtilityContext context, ParamListInfo params,
+					QueryEnvironment *queryEnv, DestReceiver *dest, char *completionTag);
+#define _PU_HOOK \
+static void PU_hook(PlannedStmt *pstmt, const char *queryString, \
+					ProcessUtilityContext context, ParamListInfo params, \
+					QueryEnvironment *queryEnv, \
+					DestReceiver *dest, char *completionTag)
+
+#define _prev_hook \
+	prev_hook(pstmt, queryString, context, params, queryEnv, dest, completionTag)
+
+#define _standard_ProcessUtility \
+		standard_ProcessUtility(pstmt, queryString, context, params, queryEnv, dest, completionTag)
+
+# endif
+
+#include "utils/varlena.h"
+#define HAS_PSTMT
```

Comment:
this is cruft. it can be dropped.

Reply:
i'll hold off for further reviews and fix this in the respin with whatever else needs to be fixed.